public/app/plugins/panel/table/TablePanel.tsx

```diff
@@ -15,16 +16,15 @@ export function TablePanel(props: Props) {
 
   const theme = useTheme2();
   const panelContext = usePanelContext();
-  const frames = data.series;
-  const mainFrames = frames.filter((f) => f.meta?.custom?.parentRowIndex === undefined);
-  const subFrames = frames.filter((f) => f.meta?.custom?.parentRowIndex !== undefined);
-  const count = mainFrames?.length;
-  const hasFields = mainFrames[0]?.fields.length;
-  const currentIndex = getCurrentFrameIndex(mainFrames, options);
-  const main = mainFrames[currentIndex];
+  const frames = hasDeprecatedParentRowIndex(data.series)
```

Comment:
Could migration happen inside Table component? I'm thinking about a use case if someone has an App plugin and uses Table component directly from grafana-ui, not TablePanel. In such case migration won't happen and the only way would be to write it manually (app plugins won't be able to import migration functions).

Reply:
Would need to move the migration logic to the Table component itself for the reasons you suggested. Feels kind of messy as migration logic for panels seems to be in the panel folder itself which seems to be the norm. I'm thinking that a subset of app plugins use the table component and a subset of those would use it with sub table. Even at that, the changes required are actually quite minimal for a user to migrate themselves (which they will eventually have to do anyway). This may essentially be a non issue, but can't be certain.